src/vs/workbench/contrib/terminalContrib/accessibility/browser/terminalAccessibleBuffer.ts

```diff
@@ -95,42 +102,61 @@ export class AccessibleBufferWidget extends DisposableStore {
 		this._xterm.raw.focus();
 	}
 
-	async show(): Promise<void> {
-		const commandDetection = this._capabilities.get(TerminalCapability.CommandDetection);
-		const fragment = !!commandDetection ? this._getShellIntegrationContent() : this._getAllContent();
-		const model = await this._getTextModel(URI.from({ scheme: AccessibleBufferConstants.Scheme, fragment }));
+	private async _updateContent(refresh?: boolean): Promise<ITextModel> {
+		const sb = new StringBuilder(10000);
+		if (refresh) {
+			sb.appendString(this._content);
+			sb.appendString(this._getContent(this._lastContentLength));
+		} else {
+			const commandDetection = this._capabilities.get(TerminalCapability.CommandDetection);
+			sb.appendString(!!commandDetection ? this._getShellIntegrationContent() : this._getContent());
+		}
+		this._content = sb.build();
```

Comment:
`_content` doesn't appear to be cleared, so I think it will become corrupted if the cursor moves up or to the left? You could leverage the fact that the `scrollback` is readonly; it can only be cleared completely.

Reply:
i think we want to clear `_content` on command finished

Reply:
Isn't `_content` just a mirror of all the content in the buffer? I didn't think it should have anything to do with shell integration now?